pipeline/builder/builder.go

```diff
@@ -319,51 +319,61 @@ func (b *Builder) buildDeployEmbeddedManifestStage(index int, s config.Stage) (*
+				} else {
+					// CronJob nested struct
+					c, _, _ = unstructured.NestedFieldNoCopy(u.Object, "spec", "jobTemplate", "spec", "template", "spec", "containers")
+				}
 
-					// set resources limits
-					limits, _, _ := unstructured.NestedFieldNoCopy(c, "resources", "limits")
-					limitsTyped, err := toResourceList(limits)
-					if err != nil {
-						return nil, errors.Wrapf(err, "failed to convert resources limits to a resource list for container: %s", overrideContainer.Name)
-					}
-					limits, err = overrideResource(limitsTyped, overrideContainer.Resources.Limits)
-					if err != nil {
-						return nil, errors.Wrapf(err, errOverrideResource, "limits", overrideContainer.Name)
-					}
-					if err := setNestedFieldNoCopy(c, limits, "resources", "limits"); err != nil {
-						return nil, errors.Wrapf(err, "failed to set resources requests for container: %s", overrideContainer.Name)
+				containers := c.([]interface{})
+
+				for i, unstructuredContainer := range containers {
+					container := unstructuredContainer.(map[string]interface{})
+					for _, overrideContainer := range maniStage.ContainerOverrides {
+						if container["name"] != overrideContainer.Name || overrideContainer.Resources == nil {
+							continue
+						}
+						c := (containers[i]).(map[string]interface{})
+						// set resources requests
+						requests, _, _ := unstructured.NestedFieldNoCopy(c, "resources", "requests")
+						requestsTyped, err := toResourceList(requests)
+						if err != nil {
+							return nil, errors.Wrapf(err, "failed to convert resources requests to a resource list for container: %s", overrideContainer.Name)
+						}
+						requests, err = overrideResource(requestsTyped, overrideContainer.Resources.Requests)
+						if err != nil {
+							return nil, errors.Wrapf(err, errOverrideResource, "requests", overrideContainer.Name)
+						}
+						if err := setNestedFieldNoCopy(c, requests, "resources", "requests"); err != nil {
+							return nil, errors.Wrapf(err, "failed to set resources requests for container: %s", overrideContainer.Name)
+						}
+
+						// set resources limits
+						limits, _, _ := unstructured.NestedFieldNoCopy(c, "resources", "limits")
+						limitsTyped, err := toResourceList(limits)
+						if err != nil {
+							return nil, errors.Wrapf(err, "failed to convert resources limits to a resource list for container: %s", overrideContainer.Name)
+						}
+						limits, err = overrideResource(limitsTyped, overrideContainer.Resources.Limits)
+						if err != nil {
+							return nil, errors.Wrapf(err, errOverrideResource, "limits", overrideContainer.Name)
+						}
+						if err := setNestedFieldNoCopy(c, limits, "resources", "limits"); err != nil {
+							return nil, errors.Wrapf(err, "failed to set resources requests for container: %s", overrideContainer.Name)
+						}
 					}
 				}
+				objs[i] = u
+			} else {
+				continue
```

Comment:
not sure if convention or if we should just get rid of this `else continue`

Reply:
100% correct